pkg/controller/subscription/rhmiConfigs/main.go

```diff
@@ -69,13 +70,60 @@ func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integrea
 		config.Status.Maintenance.Duration = strconv.Itoa(WINDOW) + "hrs"
 	}
 
+	// Calculate the upgrade window
 	if installplan.Spec.Approved {
 		config.Status.Upgrade.Window = ""
 	} else {
 		upStart := installplan.ObjectMeta.CreationTimestamp.Time.Format("2 Jan 2006")
-		upEnd := installplan.ObjectMeta.CreationTimestamp.Time.Add((time.Hour * 24) * 14).Format("2 Jan 2006")
+		upEnd := installplan.ObjectMeta.CreationTimestamp.Time.Add(daysDuration(14)).Format("2 Jan 2006")
 		config.Status.Upgrade.Window = upStart + " - " + upEnd
 	}
+
+	// Calculate the upgrade schedule based on the spec:
+	// * If it's set to AlwaysImmediatly, there's no schedule
+	if config.Spec.Upgrade.AlwaysImmediately {
+		config.Status.Upgrade.Scheduled = nil
+
+	} else
+	// * If it's set to next maintenance, it's scheduled at the next maintenance window
+	if config.Spec.Upgrade.DuringNextMaintenance {
```

Comment:
```suggestion
	} 
	// * If it's set to next maintenance, it's scheduled at the next maintenance window
	else if config.Spec.Upgrade.DuringNextMaintenance {
```
```suggestion
	} 
	else if config.Spec.Upgrade.DuringNextMaintenance {
	// * If it's set to next maintenance, it's scheduled at the next maintenance window
	
```
Its a bit hard to read for me that it was a `else if` due to the comment in between. Either moving the `else` to the same line to have the comment above the statement or moving the comment below the statement helps with readability I think :thinking:

Reply:
I'm not a big fan of that style, the only reason I used it was because VSCode would force me to indent the comment otherwise, but if it hurts readability I'll have a look at your suggestions

Reply:
Oh okay, if it's a conventional style I'm good with it, just seemed a bit weird for me to see an `else if` broken with a comment in between, might be just me though :sweat_smile:

Reply:
Oh no, I totally agree with you. I wouldn't have used that style if it wasn't for VS Code forcing the indentation on the comments above, but I didn't notice that it would affect readability, so I changed it back to `else if` together as you suggested